File: 2476-check-distances-between-same-letters/check-distances-between-same-letters.js
/**
 * @param {string} s
 * @param {number[]} distance
 * @return {boolean}
 */
var checkDistances = function(s, distance) {
    let d = {}
    for (let i=0;i<s.length;i++){
        if(!(s[i] in d)){
            d[s[i]] = i
        }else{
            d[s[i]] = Math.abs(i - d[s[i]])-1
        }
    }
    for(let i in d){
        curr = i.charCodeAt(0) - "a".charCodeAt(0)
        if(distance[curr] != d[i] ){
            return false
        }
    }
    return true
};

File: 2476-check-distances-between-same-letters/check-distances-between-same-letters.py
class Solution:
    def checkDistances(self, s: str, distance: List[int]) -> bool:
        d = {}
        for i in range(len(s)):
            if s[i] not in d:
                d[s[i]] = i
            else:
                d[s[i]] = abs( i - d[s[i]])-1
        for i in d:
            cur = ord(i) - ord("a")
            if distance[cur] != d[i]:
                return False
        return True